mne_neuron/dipole.py

```diff
@@ -14,55 +14,109 @@ def _hammfilt(x, winsz):
+    err0 = rmse(ddat['dextdata'][:,1], dpldown)
+    return err0
+
+
+def initialize_sim_once(net):
+    """
+    Initialize NEURON simulation variables
 
     Parameters
     ----------
     net : Network object
-        The Network object specifying how cells are
-        connected.
-    n_jobs : int
-        The number of jobs to run in parallel
-
-    Returns
-    -------
-    dpl: instance of Dipole
-        The dipole object
+        The Network object with parameter values
     """
-    from .parallel import rank, nhosts, pc, cvode
-    from .network import Network
 
+    from .parallel import pc
     from neuron import h
     h.load_file("stdrun.hoc")
 
-    net = Network(net.params, n_jobs=n_jobs)
+    global t_vec, dp_rec_L2, dp_rec_L5
 
-    # Now let's simulate the dipole
-    if rank == 0:
-        print("running on %d cores" % nhosts)
+    t_vec = h.Vector()
+    dp_rec_L2 = h.Vector()
+    dp_rec_L5 = h.Vector()
 
     # global variables, should be node-independent
     h("dp_total_L2 = 0.")
     h("dp_total_L5 = 0.")
 
+    t_vec.record(h._ref_t)  # time recording
+    dp_rec_L2.record(h._ref_dp_total_L2)  # L2 dipole recording
+    dp_rec_L5.record(h._ref_dp_total_L5)  # L5 dipole recording
+
     # Set tstop before instantiating any classes
     h.tstop = net.params['tstop']
     h.dt = net.params['dt']  # simulation duration and time-step
     h.celsius = net.params['celsius']  # 37.0 - set temperature
 
-    # We define the arrays (Vector in numpy) for recording the signals
-    t_vec = h.Vector()
-    t_vec.record(h._ref_t)  # time recording
-    dp_rec_L2 = h.Vector()
-    dp_rec_L2.record(h._ref_dp_total_L2)  # L2 dipole recording
-    dp_rec_L5 = h.Vector()
-    dp_rec_L5.record(h._ref_dp_total_L5)  # L5 dipole recording
 
-    net.movecellstopos()  # position cells in 2D grid
+def simulate_dipole(net, trial=0, inc_evinput=0.0, print_progress=True, extdata=None):
+    """Simulate a dipole given the experiment parameters.
 
-    # sets the default max solver step in ms (purposefully large)
-    pc.set_maxstep(10)
+    Parameters
+    ----------
+    net : Network object
+        The Network object specifying how cells are
+        connected.
+
+    trial : int
+        Current trial number
+
+    evinputinc : float
+        An increment (in milliseconds) that gets added
+        to the evoked inputs on each successive trial.
+        The default value is 0.0.
+
+    print_progress : bool
+        False will turn off "Simulation time" messages
+
+    extdata : np.Array | None
+        Array with preloaded data to compare simulation
+        results against
+
+    Returns
+    -------
+    dpl: instance of Dipole
+        The dipole object
+    """
+    from .parallel import rank, nhosts, pc, cvode
+    from .network import Network
+
+    from neuron import h
+    h.load_file("stdrun.hoc")
+
+    # maintain vectors across trials
+    global t_vec, dp_rec_L2, dp_rec_L5
```

Comment:
this seems dangerous to me ... could we avoid the "global"

Reply:
yes, I will be updating this in a new PR for multiple trial support